share/picongpu/tests/Pusher/lib/python/test/Pusher/MainTest.py

```diff
@@ -0,0 +1,325 @@
+    unit_speed =  i.get_attribute("unit_speed")
+    unit_mass = i.get_attribute("unit_mass")
+
+    return longer_series, series, [
+        cell_depth, cell_height, cell_width, dt, unit_time, unit_charge,
+        unit_length, unit_mass, unit_speed, unit_bfield
+        ]
+
+def read_series(series):
+    """reads the position_offset (grid_knot) and position
+    (float in the intervall [0,1), relativ position in the cell) and the
+    momentum of the particle in the x-y-plane.
+
+    returns:
+        x/y_poss... array of all relative in-Cell positions
+        x/y_offSet... array of all knot-positions
+        x/y_momentum ... array of all momenta
+        charge... charge of particle
+        mass... mass of particle
+        num_iter... number of iterations in the provided series
+        """
+
+    num_iter = len(series.iterations)
+
+    # create arrays for saving the data
+
+    # in-cell position
+    x_poss = np.zeros(num_iter)
+    y_poss = np.zeros(num_iter)
+
+    # cells
+    x_offSet = np.zeros(num_iter)
+    y_offSet = np.zeros(num_iter)
+
+    # momentum
+    x_momentum = np.zeros(num_iter)
+    y_momentum = np.zeros(num_iter)
+
+    i = series.iterations[0]
+    # charge
+    charge = i.particles["e"]["charge"][io.Mesh_Record_Component.SCALAR]
+    # mass
+    mass = i.particles["e"]["mass"][io.Mesh_Record_Component.SCALAR]
+    
+    # for calculation of the openPMD-periode for later use
+    indices = []
+    for iteration_index in series.iterations:
+        indices.append(iteration_index)
+
+    periode = indices[1] -indices[0]
+
+    for iteration in range(0,num_iter,1):
+        i = series.iterations[iteration*periode]  # current position
+
+        # node coordinates
+        pos2_x = i.particles["e"]["position"]["x"].load_chunk()
+        pos2_y = i.particles["e"]["position"]["y"].load_chunk()
+
+        # InCell coordinates
+        pos_off2_x = i.particles["e"]["positionOffset"]["x"].load_chunk()
+        pos_off2_y = i.particles["e"]["positionOffset"]["y"].load_chunk()
+
+        p_x = i.particles["e"]["momentum"]["x"].load_chunk()
+        p_y = i.particles["e"]["momentum"]["y"].load_chunk()
+
+        # Spuehlen nicht vergessen!
+        series.flush()
+
+        # save the read data in arrays for later usage
+        x_poss[iteration] = pos2_x
+        y_poss[iteration] = pos2_y
+
+        x_offSet[iteration] = pos_off2_x
+        y_offSet[iteration] = pos_off2_y
+
+        x_momentum[iteration] = p_x
+        y_momentum[iteration] = p_y
+
+    del series
+    return (x_poss, y_poss, x_offSet, y_offSet, x_momentum, y_momentum,
+            charge, mass, num_iter, periode)
+
+
+def compare_momentum(x_momentum, y_momentum,
+                     charge, mass, B, params, num_iter, periode):
+    """
+    tests if the change in radius from one revolution to the other is
+    greater than epsilon = 1e-5
+
+    the radii are calculated by the momenta in the x-y-plane (and q*B)
+
+    if this statement ist true: return 0
```

Comment:
please add a description of the function arguments

Reply:
done